ecs-cli/modules/cli/compose/project/project_parseV1V2_test.go

```diff
@@ -0,0 +1,344 @@
+package project
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/compose/containerconfig"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseV1V2_Version1_HappyPath(t *testing.T) {
+	// expected ContainerConfig values
+	redisImage := "redis"
+	webImage := "web"
+	cpuShares := int64(73)
+	command := []string{"bundle exec thin -p 3000"}
+	dnsServers := []string{"1.2.3.4"}
+	dnsSearchDomains := []string{"search.example.com"}
+	entryPoint := []string{"/code/entrypoint.sh"}
+	env := []*ecs.KeyValuePair{
+		{
+			Name:  aws.String("RACK_ENV"),
+			Value: aws.String("development"),
+		},
+	}
+	extraHosts := []*ecs.HostEntry{
+		{
+			Hostname:  aws.String("test.local"),
+			IpAddress: aws.String("127.10.10.10"),
+		},
+	}
+	hostname := "foobarbaz"
+	labels := map[string]*string{
+		"label1":         aws.String(""),
+		"com.foo.label2": aws.String("value"),
+	}
+	links := []string{"redis:redis"}
+	logDriver := aws.String("json-file")
+	logOpts := map[string]*string{
+		"max-file": aws.String("50"),
+		"max-size": aws.String("50k"),
+	}
+	memory := int64(953) // 1000000000 in Mib
+	mountPoints := []*ecs.MountPoint{
+		{
+			ContainerPath: aws.String("./code"),
+			ReadOnly:      aws.Bool(false),
+			SourceVolume:  aws.String("volume-0"),
+		},
+	}
+	ports := []*ecs.PortMapping{
+		{
+			ContainerPort: aws.Int64(5000),
+			HostPort:      aws.Int64(5000),
+			Protocol:      aws.String("tcp"),
+		},
+		{
+			ContainerPort: aws.Int64(8001),
+			HostPort:      aws.Int64(8001),
+			Protocol:      aws.String("tcp"),
+		},
+	}
+	privileged := true
+	readonly := true
+	securityOpts := []string{"label:type:test_virt"}
+	shmSize := int64(128)
+	user := "user"
+	workingDir := "/var"
+
+	composeV1FileString := `web:
```

Comment:
can we add yaml args & conversion checks for `volumes_from`,  `cap_add`  and `cap_drop`, since these fields are present in V1? (unless libcompose doesn't parse these for V1?)
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-compose.html
https://docs.docker.com/compose/compose-file/compose-file-v1/#cap_add-cap_drop

Reply:
Good catch on volumes. I can add that.

I didn't add `cap_add` and `cap_drop` here since this compose file string was lifted from the old test in `project_test.go`. I added it in the v2 test since it wasn't being checked and we added support for those fields when v2 was already out and more widely used.

We can maybe discuss testing strategy in person, bc initially I wanted to mostly port over the existing tests. Not sure how worthwhile it is to re-assert on fields/values shared in both v1 and v2 here.

Reply:
Agreed, let's discuss our expectation re: testing offline. I see your point about re-asserting on V1 vs. V2 output - though we should cover all the fields we support, it's probably not worth getting too fine-grained with the differences b/c at that point we're just testing the libcompose output (which we expect to handle v1/2 parsing diffs)